docs/subnets/deploy-a-gnosis-safe-on-your-evm.md

```diff
@@ -0,0 +1,827 @@
+
+Some use cases may apply such as:
+- `Upgrading a Vault` - Point the proxy implementation to a new treasury smart contract.
+- `Upgrading a Registry` - Migrate a database of user addresses and privileges to a new smart contract. This may include Owners, Stakers, Validators, Token holders, Whitelists. 
+
+### Setup
+For this part of the tutorial, we will need to clone the [Avalanche Smart Contract Quickstart repository](https://github.com/ava-labs/avalanche-smart-contract-quickstart/tree/proxy-contract-implementation) and switch to the `proxy-contract-implementation` branch the by running the following Commands:
+
+```zsh
+git clone https://github.com/ava-labs/avalanche-smart-contract-quickstart
+cd avalanche-smart-contract-quickstart
+git switch proxy-contract-implementation 
+yarn
+```
+
+Next, implement the environment and network setup [above](#setup) to prepare the Proxy Smart Contract project.
+
+### Deploy the proxy
+
+Let's deploy the proxy contracts by running the following command:
+
+```zsh
+npx hardhat run --network subnet scripts/deployStorage.ts
+```
+
+Output:
+```zsh
+Deploying Storage...
+Storage deployed to: 0x5dda6Fa725248D95d2086F4fcEb6bA6bdfEbc45b
+{ number: '42' }
+```
+
+This command actually executed 3 operations:
+- Deployed a Proxy Admin contract and assigned the deployer's address as the owner
+- Deployed the `Storage` contract and set the `number` to 42
+- Deployed a Transparent upgradeable proxy and added the `Storage` contract's address as its `implementation`
+
+Notice the line _Storage deployed to:_ in our deployment output includes the address `0x5dda6Fa725248D95d2086F4fcEb6bA6bdfEbc45b`
+
+This is our proxy address which you can also find in `.openzeppelin`, a session file that includes all of the project's proxy contract information.
+
+Example:
+```json
+{
+  "manifestVersion": "3.2",
+  "admin": {
+    "address": "0xd8215b138ef5eA0ecFc49fBaD1a30A18a109A06c",
+    "txHash": "0xf0457a8ca950fe526cc9d60fb578761538d037ea2f939758c6810a3b1e6b95d4"
+  },
+  "proxies": [
+    {
+      "address": "0x5dda6Fa725248D95d2086F4fcEb6bA6bdfEbc45b",
+      "txHash": "0x10480134bfe1709277e4e03aeed2825355c87f6a35633c6ed06a114fc9ce06a6",
+      "kind": "transparent"
+    }
+  ],
+  "impls": {
+    "cba9f8cf52e3c449631a04ea218a6cedcaf7c366669cfc257c89a008266c768f": {
+      "address": "0x42420054623f00CE5F04Ae4Fb8905f3Dd04DD27a",
+      "txHash": "0x6494655b779015d7cac8f32b7fa1d6437616de71e71312cbf17cf9cc1054ea35",
+      "layout": {
+        "storage": [
+          {
+            "label": "number",
+            "offset": 0,
+            "slot": "0",
+            "type": "t_uint256",
+            "contract": "Storage",
+            "src": "contracts/Storage.sol:7"
+          }
+        ],
+        "types": {
+          "t_uint256": {
+            "label": "uint256",
+            "numberOfBytes": "32"
+          }
+        }
+      }
+    }
+  }
+}
+``` 
+
+### Transfer The Proxy Admin Role To A Safe
```

Comment:
Please check the title case with https://titlecase.com/ and all other section titles.